samples/boot/minimal/src/test/java/sample/ClientCredentialsAuthenticationFilterTest.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sample;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.mock.web.MockFilterChain;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.server.authorization.authentication.OAuth2ClientAuthenticationToken;
+import org.springframework.util.Assert;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Base64.getEncoder;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+
+public class ClientCredentialsAuthenticationFilterTest {
+
+	public static final String CLIENT_ID = "myclientid";
+	public static final String CLIENT_SECRET = "myclientsecret";
+
+	private final ClientCredentialsAuthenticationFilter filter = new ClientCredentialsAuthenticationFilter();
+	private final AuthenticationManager authenticationManager = authentication -> {
+		Assert.isInstanceOf(OAuth2ClientAuthenticationToken.class, authentication);
+		OAuth2ClientAuthenticationToken token = (OAuth2ClientAuthenticationToken) authentication;
+		if (CLIENT_ID.equals(token.getClientId()) && CLIENT_SECRET.equals(token.getClientSecret())) {
+			authentication.setAuthenticated(true);
+			return authentication;
+		}
+		throw new BadCredentialsException("Bad credentials");
+	};
+
+
+	@BeforeEach
+	public void setup() {
+		filter.setAuthenticationManager(authenticationManager);
+		SecurityContextHolder.clearContext();
+	}
+
+	@Test
+	public void endpointUriMismatch() throws Exception {
```

Comment:
Please rename test methods to follow the convention `methodWhenConditionThenResult` . See [OAuth2AuthorizationEndpointFilterTests](https://github.com/spring-projects-experimental/spring-authorization-server/blob/master/core/src/test/java/org/springframework/security/oauth2/server/authorization/web/OAuth2AuthorizationEndpointFilterTests.java#L61) for examples.